conformance_pack/ssm.sp

```diff
@@ -68,3 +68,77 @@ control "ssm_managed_instance_compliance_patch_compliant" {
     soc_2                  = "true"
   })
 }
+
+query "ec2_instance_ssm_managed" {
+  sql = <<-EOQ
+    select
+      -- Required Columns
+      i.arn as resource,
+      case
+        when i.instance_state = 'stopped' then 'info'
+        when m.instance_id is null then 'alarm'
+        else 'ok'
+      end as status,
+      case
+        when i.instance_state = 'stopped' then i.title || ' is in stopped state.'
+        when m.instance_id is null then i.title || ' not managed by AWS SSM.'
+        else i.title || ' managed by AWS SSM.'
+      end as reason
+      -- Additional Dimentions
+      ${local.tag_dimensions_sql}
+      ${replace(local.common_dimensions_qualifier_sql, "__QUALIFIER__", "i.")}
+    from
+      aws_ec2_instance i
+      left join aws_ssm_managed_instance m on m.instance_id = i.instance_id;
+  EOQ
+}
+
+query "ssm_managed_instance_compliance_association_compliant" {
+  sql = <<-EOQ
+    select
+      -- Required Columns
+      id as resource,
+      case
+        when c.status = 'COMPLIANT' then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when c.status = 'COMPLIANT' then c.resource_id || ' association ' || c.title || ' is compliant.'
+        else c.resource_id || ' association ' || c.title || ' is non-compliant.'
+      end as reason
+      -- Additional Dimensions
+      ${local.tag_dimensions_sql}
+      ${replace(local.common_dimensions_qualifier_sql, "__QUALIFIER__", "c.")}
+    from
+      aws_ssm_managed_instance as i,
+      aws_ssm_managed_instance_compliance as c
+    where
+      c.resource_id = i.instance_id
+      and c.compliance_type = 'Association';
+  EOQ
+}
+
+query "ssm_managed_instance_compliance_patch_compliant" {
+  sql = <<-EOQ
+    select
+      -- Required Columns
+      id as resource,
+      case
+        when c.status = 'COMPLIANT' then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when c.status = 'COMPLIANT' then c.resource_id || ' patch ' || c.title || ' is compliant.'
+        else c.resource_id || ' patch ' || c.title || ' is non-compliant.'
+      end as reason
+      -- Additional Dimensions
+      ${local.tag_dimensions_sql}
```

Comment:
@madhushreeray30 ${local.tag_dimensions_sql} should not be here, as the respective tables not supporting tags. Pls check below outcome when executed with `tag_dimensions` in spvars
```
| + 2 All EC2 instances managed by Systems Manager should be compliant with patching requirements ............................................................. HIGH     1 /     1 [=         ]
| | 
| | ERROR: column "tags" does not exist (SQLSTATE 42703)
| | 
| + 3 Instances managed by Systems Manager should have an association compliance status of COMPLIANT ...............................................................     1 /     1 [=         ]
|   
|   ERROR: column "tags" does not exist (SQLSTATE 42703)
|   
```